### PR TITLE
Feature/paraphrasing training

### PR DESCRIPTION
**Pull request by erees1**

### CODE REVIEW

File: cot_transparency/apis/openai/finetune.py [hunk 1]
```diff
@@ -295,15 +329,23 @@ def run_finetune(
     write_jsonl_path.parent.mkdir(parents=True, exist_ok=True)
     # write to file
     write_jsonl_file_from_basemodel(path=write_jsonl_path, basemodels=samples)
+
+    if val_samples is not None:
+        val_file_name = Path(f"./data/uploaded_finetuning_files/{params.model}-{now_time}-val.jsonl")
+        write_jsonl_file_from_basemodel(path=val_file_name, basemodels=val_samples)
+    else:
+        val_file_name = None
```
Comment (thejaminator) [inline cot_transparency/apis/openai/finetune.py:337]: nit: i think you could avoid weird potential bugs when someone passes an empty list
```
val_samples: Sequence[FinetuneSample] = [],
```
and you'll just do 

```
    if val_samples:
        val_file_name = Path(f"./data/uploaded_finetuning_files/{params.model}-{now_time}-val.jsonl")
        write_jsonl_file_from_basemodel(path=val_file_name, basemodels=val_samples)
    else:
        val_file_name = None
````

File: cot_transparency/data_models/models.py [hunk 2]
```diff
@@ -39,6 +39,25 @@ class BaseTaskSpec(HashableBaseModel):
...
+    def get_data_example_obj(self) -> DataExampleBase:
+        raise NotImplementedError
+
+    @abstractmethod
+    def copy_update(
+        self,
+        *,
+        messages: Sequence[ChatMessage] | Unset = _UNSET,
+    ) -> "BaseTaskSpec":
+        raise NotImplementedError
```
Comment (thejaminator) [inline cot_transparency/data_models/models.py:55]: oh, does making it params only help to not break liskov? cool

File: cot_transparency/apis/openai/finetune.py [hunk 3]
```diff
@@ -236,34 +243,60 @@ def queue_finetune(file_id: str, model: str, hyperparameters: FineTuneHyperParam
...
     )
     file_id = file_upload_resp["id"]
+    if val_file_path:
+        val_samples = read_jsonl_file_into_basemodel(val_file_path, FinetuneSample)
+        val_buffer = create_openai_buffer(val_samples)
+        val_file_upload_resp: dict[str, Any] = openai.File.create(  # type: ignore[reportGeneralTypeIssues]
+            file=val_buffer,
+            purpose="fine-tune",
+            user_provided_filename=str(val_file_path),
+        )
```
[the remark this answers is not in the document]
Comment (erees1) [reply cot_transparency/apis/openai/finetune.py:283]: Yep it has to be fine-tune. And it works (at least is showing up on the OA interface)

image

File: cot_transparency/apis/openai/finetune.py (hunk 1 above)
Comment (erees1) [reply cot_transparency/apis/openai/finetune.py:337]: Not sure I follow on this one? You mean make the default an empty list?

File: cot_transparency/data_models/models.py (hunk 2 above)
Comment (erees1) [reply cot_transparency/data_models/models.py:55]: Yeah this helped me override the method and not have complaints about overriding in an incompatible manner due to positional args.

Comment (thejaminator) [reply cot_transparency/data_models/models.py:55]: smart

File: cot_transparency/apis/openai/finetune.py (hunk 1 above)
Comment (thejaminator) [reply cot_transparency/apis/openai/finetune.py:337]: yeah, when looking at Optional[list[Something]], the question is: is there a difference between None and an Empty list? if there isn't,  we can  just make it list[Something] instead.

**Pull request merged**